packages/components/src/button/index.jsx

```diff
@@ -28,16 +29,18 @@ export default class Button extends PureComponent {
 	};
 
 	render() {
-		const className = classNames( 'button', this.props.className, {
-			'is-compact': this.props.compact,
-			'is-primary': this.props.primary,
-			'is-scary': this.props.scary,
-			'is-busy': this.props.busy,
-			'is-borderless': this.props.borderless,
-		} );
+		const className = this.props.bare
+			? classNames( 'button-bare', this.props.className )
+			: classNames( 'button', this.props.className, {
+					'is-compact': this.props.compact,
+					'is-primary': this.props.primary,
+					'is-scary': this.props.scary,
+					'is-busy': this.props.busy,
+					'is-borderless': this.props.borderless,
+			  } );
```

Comment:
We might want to warn that the `bare` and `(compact|primary|scary|...)` properties are mutually exclusive. Adding them should be harmless, but they won't do anything, as all the styles are scoped as `.button.is-compact` etc.